primitives/core/src/lib.rs

```diff
@@ -403,38 +404,62 @@ pub const MAX_POSSIBLE_ALLOCATION: u32 = 33554432; // 2^25 bytes, 32 MiB
 #[rustfmt::skip]
 macro_rules! generate_feature_enabled_macro {
 	( $macro_name:ident, $feature_name:meta, $d:tt ) => {
-		/// Enable/disable the given code depending on
-		#[doc = concat!("`", stringify!($feature_name), "`")]
-		/// being enabled for the crate or not.
-		///
-		/// # Example
-		///
-		/// ```nocompile
-		/// // Will add the code depending on the feature being enabled or not.
-		#[doc = concat!(stringify!($macro_name), "!( println!(\"Hello\") )")]
-		/// ```
-		#[cfg($feature_name)]
-		#[macro_export]
-		macro_rules! $macro_name {
-			( $d ( $d input:tt )* ) => {
-				$d ( $d input )*
+		$crate::paste::paste!{
+			/// Enable/disable the given code depending on
+			#[doc = concat!("`", stringify!($feature_name), "`")]
+			/// being enabled for the crate or not.
+			///
+			/// # Example
+			///
+			/// ```nocompile
+			/// // Will add the code depending on the feature being enabled or not.
+			#[doc = concat!(stringify!($macro_name), "!( println!(\"Hello\") )")]
+			/// ```
+			#[cfg($feature_name)]
+			#[macro_export]
+			macro_rules! [<_ $macro_name>] {
+				( $d ( $d input:tt )* ) => {
+					$d ( $d input )*
+				}
+			}
+
+			/// Enable/disable the given code depending on
+			#[doc = concat!("`", stringify!($feature_name), "`")]
+			/// being enabled for the crate or not.
+			///
+			/// # Example
+			///
+			/// ```nocompile
+			/// // Will add the code depending on the feature being enabled or not.
+			#[doc = concat!(stringify!($macro_name), "!( println!(\"Hello\") )")]
+			/// ```
+			#[cfg(not($feature_name))]
+			#[macro_export]
+			macro_rules! [<_ $macro_name>] {
+				( $d ( $d input:tt )* ) => {};
 			}
-		}
 
-		/// Enable/disable the given code depending on
-		#[doc = concat!("`", stringify!($feature_name), "`")]
-		/// being enabled for the crate or not.
-		///
-		/// # Example
-		///
-		/// ```nocompile
-		/// // Will add the code depending on the feature being enabled or not.
-		#[doc = concat!(stringify!($macro_name), "!( println!(\"Hello\") )")]
-		/// ```
-		#[cfg(not($feature_name))]
-		#[macro_export]
-		macro_rules! $macro_name {
-			( $d ( $d input:tt )* ) => {};
+			// Work around for: <https://github.com/rust-lang/rust/pull/52234>
+			#[doc(hidden)] 
+			pub use [<_ $macro_name>] as $macro_name;
```

Comment:
Could you maybe sanity check these macro changes here @bkchr ?  
The other stuff is trivial.

Reply:
WDYT @bkchr?